### ISSUE TITLE
Merge with on column condition

### ISSUE THREAD

**Issue by dagarshali** [man-group/dtale#642]
The example pane for merge shows that you can merge, say left join with a common column as the join condition. However, that is not what shows up the code that you see when you add the two dataframes, you see left=index and right=index. How do you change this behavior?

**Comment by aschonfeld** [comment]
Not sure why you would want to change it.  This code should work (FYI: I did fix some small frontend bugs that I'll release soon): 
```
import dtale
from dtale.views import startup

df1 = dtale.get_instance('1').data.set_index(['category_val'])
df2 = dtale.get_instance('1').data.set_index(['Col0'])
final_df = df1.merge(df2, how='left', left_index=True, right_index=True)
startup(final_df)
```

![image](https://user-images.githubusercontent.com/11547371/157879628-89fbbc45-3d94-4724-99ed-ec6c2027201b.png)

The difference with setting the column names versus saying `left_index=True, right_index=True` is that I'm turning the selected columns into indexes before using them in a merge.  Which should be the same as passing them in manually.

**Comment by aschonfeld** [comment]
@dagarshali just released v2.1.0 with some fixes.  Please let me know if you're still having issues.  Also, if you haven't already, please put your ⭐  on the repo. Thanks 🙏
